### ISSUE TITLE
Gotify

### ISSUE THREAD

**Issue by tobru** [healthchecks/healthchecks#270]
It would be great to have [Gotify](https://gotify.net/) integration out-of-the-box in Healthchecks.io.

**Comment by baudneo** [comment]
Is Gotify still being considered for integration?

**Comment by setop** [comment]
I have an integration with Gotify using webhook :

```
URL : https://gotify.provider.tld/message?token=THEAPPLICATIONTOKEN
Method : POST
Request Body (down) : title=healthcheck&message=$NAME+$STATUS&priority=8
Request Body (up) : title=healthcheck&message=$NAME+$STATUS&priority=7
Request Headers: Content-Type: application/x-www-form-urlencoded
```

The trick to make it work is the `Content-Type: application/x-www-form-urlencoded` header.
Also note I choose priority 8 for down and 7 for up, it changes the notification kind on Android (8 is more intrusive than 7).

**Comment by cuu508** [comment]
Gotify accepts JSON payload too:

```
URL : https://gotify.provider.tld/message?token=THEAPPLICATIONTOKEN
Method : POST
Request Body (down) : {"message": "$NAME is DOWN"}
Request Body (up) : {"message": "$NAME is UP"}
Request Headers: Content-Type: application/json
```

I'll look into a native integration, it would enable nicer notifications with more content.

From a glance at Gotify docs, I'm not sure how to handle the priority values.

* should Healthchecks include `priority` key in the notification payload?
* should Healthchecks let the user configure the priority values for "down" and "up" events?
* if yes, are there human-friendly descriptions of the different priority values (e.g. "1" - Silent, "10" - "Emergency")?

**Comment by cuu508** [comment]
The initial version of Gotify integration is ready, and available on https://healthchecks.io

**Comment by sagehou** [comment]
The experience on [the website](https://healthchecks.io)https://healthchecks.io/ is good. Hope the function can be added to docker images.

**Comment by cuu508** [comment]
@sagehou I released v2.2.1 today, updated Docker images are now available

**Comment by setop** [comment]
I change my config from webhook to native integration. Test is ok. Waiting for the next failure to fully confirm :)
For sure the logo is nice :)

**Comment by sagehou** [comment]
> @sagehou I released v2.2.1 today, updated Docker images are now available

Nice job 👍 .
